crates/ra_assists/src/assists/invert_if.rs

```diff
@@ -0,0 +1,89 @@
+use hir::db::HirDatabase;
+use ra_syntax::ast::{self, AstNode};
+use ra_syntax::{TextRange, TextUnit};
+
+use super::apply_demorgan::undo_negation;
+use crate::{Assist, AssistCtx, AssistId};
+
+// Assist: invert_if
+//
+// Apply invert_if
+// This transforms if expressions of the form `if !x {A} else {B}` into `if x {B} else {A}`
+// This also works with `!=`. This assist can only be applied with the cursor
+// on `if`.
+//
+// ```
+// fn main() {
+//     if<|> !y {A} else {B}
+// }
+// ```
+// ->
+// ```
+// fn main() {
+//     if y {B} else {A}
+// }
+// ```
+
+pub(crate) fn invert_if(ctx: AssistCtx<impl HirDatabase>) -> Option<Assist> {
+    let expr = ctx.find_node_at_offset::<ast::IfExpr>()?;
+    let expr_range = expr.syntax().text_range();
+    let if_range = TextRange::offset_len(expr_range.start(), TextUnit::from_usize(2));
+    let cursor_in_range = ctx.frange.range.is_subrange(&if_range);
+    if !cursor_in_range {
+        return None;
+    }
+
+    let cond = expr.condition()?.expr()?.syntax().clone();
+    let then_node = expr.then_branch()?.syntax().clone();
+
+    if let ast::ElseBranch::Block(else_block) = expr.else_branch()? {
+        let flip_cond = undo_negation(cond.clone())?;
+        let cond_range = cond.text_range();
+        let else_node = else_block.syntax();
+        let else_range = else_node.text_range();
+        let then_range = then_node.text_range();
+        ctx.add_assist(AssistId("invert_if"), "invert if branches", |edit| {
+            edit.target(if_range);
+            edit.replace(cond_range, flip_cond);
+            edit.replace(else_range, then_node.text());
+            edit.replace(then_range, else_node.text());
```

Comment:
Another option would be to use `edit::replace_descendants` method, though I am not sure it'll work here in the current form...